### ISSUE TITLE
More .torrent file sources

### ISSUE THREAD

**Issue by duramato** [FlyersWeb/dhtbay#6]
Add the possibility  to add via a config file, ideally,  more sources to download .torrents. 
They could be added as they are atm to the config file `http://torcache.net/torrent/{DHTHASH}.torrent` , so that despite the format the only part that would change would be the DHTHASH, so it could even be `http://mydomain.com/download/torrent/{DHTHASH}`

**Comment by duramato** [comment]
I did notice that currently the "sources" are set here https://github.com/FlyersWeb/dhtbay/blob/master/loadDHT.js#L18

I did try adding more sources to [L19](https://github.com/FlyersWeb/dhtbay/blob/master/loadDHT.js#L19) , [L46](https://github.com/FlyersWeb/dhtbay/blob/master/loadDHT.js#L46) and [L50](https://github.com/FlyersWeb/dhtbay/blob/master/loadDHT.js#L50) but didn't saw any imediate effect, don't know if i did it right. 

One question: How does it handle if one source doesn't have the torrent?(a 404) moves to next torrent or tries the next sources?

**Comment by FlyersWeb** [comment]
Right now we only use torcache as a resource because torrage seemed not to work. It would be great to have more .torrent resources but as you say, they'll need to be managed dynamically to be effective because URI may change very often.
Aria2 addURI method will try to download the resource using all provided URI and if not, it just fail without downloading the .torrent.
To add more sources you'll need to add them in array on [L50](https://github.com/FlyersWeb/dhtbay/blob/master/loadDHT.js#L50)

**Comment by albabosh** [comment]
You could download torrent files with Aria2c from the DHT

**Comment by FlyersWeb** [comment]
it could be an interesting improvement but for now the project downloads from torcache

**Comment by albabosh** [comment]
It worked well for me

**Comment by duramato** [comment]
Ok this become of more importance as torcache.net is gone now.

**Comment by albabosh** [comment]
aria2c --bt-metadata-only=true [infohash] will download .torrent from dht

**Comment by duramato** [comment]
Do you know if i can make that work with dhtbay? @albabosh

**Comment by FlyersWeb** [comment]
I'm working on an improved dhtbay version using magnet instead of torcache, it's still in progress but if you want you can alpha test it @duramato.
See https://github.com/FlyersWeb/dhtbay/tree/Remove.TORCache.Refactor

**Comment by albabosh** [comment]
``` javascript
function run() {

  client.lpop("DHTS",function(err, hash){
    if(err) {console.log(err); return;}
    if(!hash) {return;}
    if(fs.existsSync(dest+hash.toString()+'.torrent')) {console.log("File "+hash.toString().toUpperCase()+".torrent already exists");return;}
    var magnet = MAGNET_TEMPLATE.replace('{DHTHASH}',hash.toString().toUpperCase());
        aria2.open(function(){
        aria2.send('addUri',[magnet],{'bt-metadata-only':'true', 'bt-save-metadata':'true', 'dir':dest}, function(err,res){
          if(err) { console.log(err); return;}
          console.log("Added : "+hash.toString().toUpperCase()+".torrent");
//          client.rpush("TORS", hash.toString());
          aria2.close();
        })
      });
    });

}
```

this is how i did it

**Comment by FlyersWeb** [comment]
Corrected by commit b6e0d24647c16b07db882b569790b46e4487ec32
